pkg/client/typed/dynamic/client.go

```diff
@@ -0,0 +1,184 @@
+		conf.Burst = 10
+	}
+
+	cl, err := client.RESTClientFor(conf)
+	if err != nil {
+		return nil, err
+	}
+
+	return &Client{cl: cl}, nil
+}
+
+// Resource returns an API interface to the specified resource for
+// this client's group and version. If resource is not a namespaced
+// resource, then namespace is ignored.
+func (c *Client) Resource(resource *unversioned.APIResource, namespace string) *ResourceClient {
+	return &ResourceClient{
+		cl:       c.cl,
+		resource: resource,
+		ns:       namespace,
+	}
+}
+
+// ResourceClient is an API interface to a specific resource under a
+// dynamic client.
+type ResourceClient struct {
+	cl       *client.RESTClient
+	resource *unversioned.APIResource
+	ns       string
+}
+
+// namespace applies a namespace to the request if the configured
+// resource is a namespaced resource. Otherwise, it just returns the
+// passed in request.
+func (rc *ResourceClient) namespace(req *client.Request) *client.Request {
+	if rc.resource.Namespaced {
+		return req.Namespace(rc.ns)
+	}
+	return req
+}
+
+// List returns a list of objects for this resource.
+func (rc *ResourceClient) List(opts api.ListOptions) (*List, error) {
+	result := new(List)
+	err := rc.namespace(rc.cl.Get()).
+		Resource(rc.resource.Name).
+		VersionedParams(&opts, api.Scheme). // What to do about VersionedParams converter?
+		Do().
+		Into(result)
+	return result, err
+}
+
+// Get gets the resource with the specified name.
+func (rc *ResourceClient) Get(name string) (*Object, error) {
+	result := new(Object)
+	err := rc.namespace(rc.cl.Get()).
+		Resource(rc.resource.Name).
+		Name(name).
+		Do().
+		Into(result)
+	return result, err
+}
+
+// Delete deletes the resource with the specified name.
+func (rc *ResourceClient) Delete(name string) error {
+	return rc.namespace(rc.cl.Delete()).
+		Resource(rc.resource.Name).
+		Name(name).
+		Do().
+		Error()
+}
+
+// Create creates the provided resource.
+func (rc *ResourceClient) Create(obj *Object) (*Object, error) {
+	result := new(Object)
+	err := rc.namespace(rc.cl.Post()).
+		Resource(rc.resource.Name).
+		Body(obj).
+		Do().
+		Into(result)
+	return result, err
+}
+
+// Update updates the provided resource.
+func (rc *ResourceClient) Update(obj *Object) (*Object, error) {
+	result := new(Object)
+	err := rc.namespace(rc.cl.Post()).
+		Resource(rc.resource.Name).
+		Name(obj.Name).
+		Body(obj).
+		Do().
+		Into(result)
+	return result, err
+}
+
+// Watch returns a watch.Interface that watches the resource.
+func (rc *ResourceClient) Watch(opts api.ListOptions) (watch.Interface, error) {
+	return rc.namespace(rc.cl.Get()).
+		Prefix("watch").
+		Resource(rc.resource.Name).
+		VersionedParams(&opts, api.Scheme). // What to do about VersionedParams converter?
+		Watch()
+}
+
+// Object is an arbitrary resource with metadata exposed.
+type Object struct {
+	unversioned.TypeMeta `json:",inline"`
+	api.ObjectMeta       `json:"metadata,omitempty"` // There's probably something better for this...
```

Comment:
Yes, this is "ListMeta for non-lists" ;)

(Consider using it from the versioned package?)

We have runtime.Unstructured and runtime.Unknown; my preference would be to reuse one of those objects and change it as necessary, it should be OK to add metadata to one of those.

May need to split into UnstructuredObject and UnstructuredList, now that I think about it...